### PR TITLE
MNT: avoid numpy.maximum_sctype (deprecated in numpy 2.0)

### PR DESCRIPTION
**Pull request by neutrinoceros**
xref https://github.com/numpy/numpy/pull/24154

### CODE REVIEW

**Comment by neutrinoceros** [comment]
honestly not sure if this warrants a changelog entry but I'm happy to add one if really needed

File: astropy/nddata/bitmask.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
 
 
 _ENABLE_BITFLAG_CACHING = True
-_MAX_UINT_TYPE = np.maximum_sctype(np.uint)
-_SUPPORTED_FLAGS = int(np.bitwise_not(0, dtype=_MAX_UINT_TYPE, casting="unsafe"))
+_SUPPORTED_FLAGS = int(np.bitwise_not(0, dtype="uint64", casting="unsafe"))
```
Comment (pllim) [inline astropy/nddata/bitmask.py:22]: But what about 32-bit machine?

Comment (neutrinoceros) [reply astropy/nddata/bitmask.py:22]: It's been discussed already upstream: https://github.com/numpy/numpy/pull/24154#discussion_r1261538812
My understanding is that `uint64` is available *even on 32-bit* architectures, and is de facto always the result of `np.maximum_sctype(np.uint)`

**Comment by pllim** [comment]
Well, the `armv7` job is supposed to represent 32-bit support and it is green, so I guess it is ok. The dev jobs are a mess right now between numpy dev and pytest dev, so we can ignore them here.

Thanks!

**Review by pllim — APPROVED** [review]
Actually I am just going to get this in. I see this downstream. Thanks for the patch!

**Pull request merged**